core/commonui/modals/src/commonMain/kotlin/com/livefast/eattrash/raccoonforlemmy/core/commonui/modals/CustomBottomSheet.kt

```diff
@@ -0,0 +1,145 @@
+package com.livefast.eattrash.raccoonforlemmy.core.commonui.modals
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Icon
+import androidx.compose.material3.ModalBottomSheet
+import androidx.compose.material3.SheetState
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.painter.Painter
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.text.style.TextAlign
+import com.livefast.eattrash.raccoonforlemmy.core.appearance.theme.Spacing
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun CustomBottomSheet(
+    isOpen: Boolean,
+    sheetState: SheetState,
+    sheetScope: CoroutineScope,
+    onDismiss: (() -> Unit)? = null,
+    onSelection: ((Int) -> Unit),
+    headerText: String,
+    contentPreTextPainter: List<Painter?>? = null,
+    contentPreTextIcon: List<ImageVector?>? = null,
+    contentText: List<String?>,
+    contentPostTextPainter: List<Painter?>? = null,
+    contentPostTextIcon: List<ImageVector?>? = null,
```

Comment:
**suggestion(non-blocking):** having 5 parameters of type list which are paired together can complicate things, e.g. when you need a `postTextIcon` for the 5th element but none for the 4th so you start inserting `null`s. Did you consider using a `data class CustomBottomSheetItem(val preTextPainter: Painter? = null, val preTextIcon: ImageVector?, val text: String? = null, val postTextPainter: Painter? = null, val postTextIcon: ImageVector? = null)` and simply pass a list of `CustomBottomSheetItem` with all the packed information?

BTW I ❤️ this idea which is great!

Reply:
Using a data class did actually cross my mind last night, but I that was before I discovered `liftOf()` :laughing: 
The thought of initializing and building the list with `list.Add()` + not knowing which file/folder to create the new data class made me mentally shelve the idea, and then I forgot about it. 
But I **definitely** want to implement that, as it makes _so_ much more sense! 

Do you think the data class should be in `CustomBottomSheet` or in a separate file?  And if separate, where would you recommend putting it? 
Thinking about this earlier, I realized that the pre icon/vector is actually not needed, so I may remove those too. 

Thank you!  I really appreciate the feedback!

Reply:
For me, it could as well be in the same file, Kotlin is not Java luckily! And since it is always used with this composable it can be placed in the same source.

There are already – alas! – such "model for the view" data classes in UI modules, the most blatant example would the content of Options.kt in `:core:commonui:lemmyui` (all models, nothing UI-specific).